web3/contract/base_contract.py

```diff
@@ -766,7 +772,7 @@ def encode_abi(
 
         fn_info = get_abi_element_info(
```

Comment:
I think this `fn_info` could be `element_info` for clarity, since we left the function filtering logic out of this method.